modal/_resolver.py

```diff
@@ -122,20 +131,33 @@ async def loader():
 
             cached_future = asyncio.create_task(loader())
             self._local_uuid_to_future[obj.local_uuid] = cached_future
+            if deduplication_key is not None:
+                self._deduplication_cache[deduplication_key] = cached_future
+
+        def hydrate_original(fut):
+            # in case an object is omitted due to content duplication, make sure the original reference
```

Comment:
The previous bug making integration tests fail was here - there was a check `if obj.is_hydrated: return` which made stale anonymous mounts from previous deploys (typically auto mounted packages) not get new ids on a second deploy...